lib/tree-view.coffee

```diff
@@ -53,7 +53,34 @@ class TreeView extends View
     @updateRoots(state.directoryExpansionStates)
     @selectEntry(@roots[0])
 
-    @selectEntryForPath(state.selectedPath) if state.selectedPath
+    # Now that directory expansion is done asyncly combining 2 different
+    # asynchronous code paths(1), there doesn't seem to be a good way to know
+    # that the tree view has been fully expanded from its initial tree of
+    # expansion states. So, we could make the expansion state construct know
+    # about when it's been fully walked, or we could just watch the DOM for the
+    # item in question to appear. I'm still a little concerned about hitting an
+    # edge case where this doesn't get ::disconnected() though.
+    #
+    # 1: Directories were expanded synchronously, but this was changed to work
+    #    with the new async git repo interaction. However, we also use event
+    #    subscription with Directory::onDidAddEntries to build up a tree view as
+    #    well, meaning we can't just tag on to the end of the Directory::expand()
+    #    promise chain :/
```

Comment:
Naively, I would think we could have `updateRoots` return a promise that resolves after all of the directories are expanded, and do something like

```coffee
@updateRoots(state.directoryExpansionStates).then =>
  if state.selectedPath
    @selectEntryForPath(state.selectedPath)
  else
    @selectEntry(@roots[0])
```

I assume I'm missing something; how does the `Directory::onDidAddEntries` code path relate to this?